test/test_random.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019-2021 Lawrence Livermore National Security, LLC and other YGM
+// Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#undef NDEBUG
+#include <ygm/comm.hpp>
+#include <ygm/container/counting_set.hpp>
+#include <ygm/random.hpp>
+
+#include <random>
+
+int main(int argc, char** argv) {
+  ygm::comm world(&argc, &argv);
+
+  //
+  // Test simpe_offset_rng
+  {
+    std::uint32_t                                seed = 100;
+    ygm::random::simple_offset_rng<std::mt19937> rng(world, seed);
```

Comment:
To keep this simple, if it was ygm::random::random_engine, it could default to std::default_random_engine.

The shared version could be:
ygm::random::shared_random_engine